src/Setup/Installer/tools/utils.ps1

```diff
@@ -72,55 +72,41 @@ function Exec-Console([string]$command, [string]$commandArgs) {
 }
 
 
-# Get the directory and instance ID of the first Visual Studio version which
+# Returns an array of JSON objects, each item representing an installed instance of Visual Studio that 
 # meets our minimal requirements for the Roslyn repo.
-function Get-VisualStudioDirAndId() {
+# Throws if there is none.
+function Get-VisualStudioInstances() {
+  $minVersionStr = "15.8"
+
   $vswhere = Join-Path $PSScriptRoot "vswhere\vswhere.exe"
-  $output = Exec-Command $vswhere "-prerelease -requires Microsoft.VisualStudio.Component.Roslyn.Compiler -version [15.5,15.9) -format json" | Out-String
-  $j = ConvertFrom-Json $output
-  $foundVsInstall = $false
-  foreach ($obj in $j) {
-    # Need to be using at least Visual Studio 15.5 in order to have the appropriate
-    # set of SDK fixes. Parsing the installationName is the only place where this is
-    # recorded in that form.
-    $name = $obj.installationName
-    if ($name -match "VisualStudio(Preview)?/([\d.]+)(\+|-).*") {
-      $minVersion = New-Object System.Version "15.5.0"
-      $maxVersion = New-Object System.Version "15.9.0"
-      $version = New-Object System.Version $matches[2]
-      if ($version -ge $minVersion -and $version -lt $maxVersion) {
-        Write-Output $obj.installationPath
-        Write-Output $obj.instanceId
-        $foundVsInstall = $true;
-      }
-    }
-  }
+  $vsInstances = Exec-Command $vswhere "-prerelease -requires Microsoft.VisualStudio.Component.Roslyn.Compiler -version $minVersionStr -format json" | ConvertFrom-Json
 
-  if (-not $foundVsInstall) {
-    throw "Could not find a suitable Visual Studio Version"
+  if ($vsInstances.Length -eq 0) {
```

Comment:
Does this mean we pick a random build (possible d16.0).  Would it be better to have a consistent algorithm for deciding what to do in the case of multiple VS installs?

Reply:
We don't pick random build. We show list of builds and let the user pick one.

Reply:
The previous impl printed out the available versions (possibly multiple of them) to std out and the caller read them from there.